yarn-project/pxe/src/pxe_service/pxe_service.ts

```diff
@@ -847,24 +857,34 @@ export class PXEService implements PXE {
         await this.synchronizer.sync();
         const syncTime = syncTimer.ms();
 
-        const contractFunctionSimulator = this.#getSimulatorForTx();
+        const contractFunctionSimulator = this.#getSimulatorForTx(overrides);
+        const skipKernels = overrides?.contracts !== undefined && Object.keys(overrides.contracts.instances).length > 0;
```

Comment:
This feels magical. Please document what's this and why is it set like that.

Reply:
This is temporary, will add a comment. There's a high degree of probability that we will make simulating without kernels the default, and provide an opt-out option once we can handle more cases (revertible/non-revertible sideeffects, squashing)